### PR TITLE
Migrate cassandra-operator from dep to go modules #60 

### PR DESCRIPTION
**Pull request by kragniz**
See #60.

`cassandra-snapshot` can be migrated separately.

### CODE REVIEW

File: cassandra-operator/hack/pin-dependency.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
```
Comment (wallrj) [inline cassandra-operator/hack/pin-dependency.sh]: Maybe add a note to the CONTRIBUTING.md document, explaining when  and how to use this script.

File: cassandra-operator/Makefile [hunk 2]
```diff
@@ -120,10 +116,7 @@ deploy-operator:
 
 validate-licences:
 	@echo "== validate licences"
-	set -e ;\
-	restricted=$$(paste -s -d ',' restricted-licences.txt) ;\
-	projects=$$(dep status -f='vendor/{{ .ProjectRoot }} ') ;\
-	licence-compliance-checker -L error -E -r $$restricted -o vendor/sigs.k8s.io/yaml=BSD-3-Clause -o vendor/github.com/golang/protobuf=BSD-3-Clause -o vendor/github.com/gogo/protobuf=BSD-3-Clause -o vendor/github.com/hashicorp/golang-lru=MPL-2.0 $$projects ;\
+	$(projectDir)/hack/check-licences.sh
```
Comment (wallrj) [inline cassandra-operator/Makefile]: I tested this by adding a GPL-3 licensed import to one of the files.
An error was reported, which is good:

```
make validate-licences
...
ERRO[0004] Some licences are not compliant and/or cannot be identified: restricted: [{/home/richard/go/pkg/mod/github.com/getgauge/common@v0.0.0-20190110102454-8966e8b0efde [{GPL-3.0-or-later 0.99014163} {deprecated_GPL-3.0 0.99014163} {GPL-3.0-only 0.99014163} {deprecated_GPL-3.0+ 0.99014163} {AGPL-3.0-or-later 0.8023801} {deprecated_AGPL-3.0 0.8023801} {AGPL-3.0-only 0.8023801}] } {/home/richard/go/pkg/mod/github.com/getgauge/gauge@v1.0.5 [{GPL-3.0-only 0.99014163} {deprecated_GPL-3.0+ 0.99014163} {GPL-3.0-or-later 0.99014163} {deprecated_GPL-3.0 0.99014163} {AGPL-3.0-or-later 0.8023801} {deprecated_AGPL-3.0 0.8023801} {AGPL-3.0-only 0.8023801}] }], unidentifiable: [] 
make: *** [Makefile:118: validate-licences] Error 1

```

**Comment by kragniz** [comment]
Change of approach, I've added support for looking up directories of go modules here, and use that feature in the `validate-licences` make targets.

This now depends on https://github.com/sky-uk/licence-compliance-checker/pull/19

File: cassandra-operator/Makefile [hunk 3]
```diff
@@ -35,15 +35,10 @@ check: check-style validate-licences test deploy-operator integration-test e2e-t
 
 setup:
 	@echo "== setup"
-	go get -v golang.org/x/lint/golint golang.org/x/tools/cmd/goimports github.com/golang/dep/cmd/dep github.com/sky-uk/licence-compliance-checker
-	go get -v github.com/onsi/ginkgo/ginkgo && cd $$GOPATH/src/github.com/onsi/ginkgo && git checkout 'v1.6.0' && go install github.com/onsi/ginkgo/ginkgo
-	curl https://raw.githubusercontent.com/golang/dep/v0.5.1/install.sh | sh
+	GO111MODULE=off go get -v golang.org/x/lint/golint golang.org/x/tools/cmd/goimports github.com/sky-uk/licence-compliance-checker
+	GO111MODULE=off go get -v github.com/onsi/ginkgo/ginkgo && cd $$GOPATH/src/github.com/onsi/ginkgo && git checkout 'v1.6.0' && GO111MODULE=off go install github.com/onsi/ginkgo/ginkgo
```
Comment (sebbonnet) [inline cassandra-operator/Makefile]: `ginkgo:v1.6.0` won't work with go modules, we will need to upgrade to `v1.7.0`

File: cassandra-snapshot/Makefile [hunk 4]
```diff
@@ -32,17 +32,10 @@ check: check-style validate-licences test prepare-deployment integration-test e2
 
 setup:
 	@echo "== setup"
-	go get -v golang.org/x/lint/golint golang.org/x/tools/cmd/goimports github.com/golang/dep/cmd/dep github.com/sky-uk/licence-compliance-checker
-	go get -v github.com/onsi/ginkgo/ginkgo && cd $$GOPATH/src/github.com/onsi/ginkgo && git checkout 'v1.6.0' && go install github.com/onsi/ginkgo/ginkgo
-	curl https://raw.githubusercontent.com/golang/dep/v0.5.1/install.sh | sh
+	GO111MODULE=off go get -v github.com/onsi/ginkgo/ginkgo && cd $$GOPATH/src/github.com/onsi/ginkgo && git checkout 'v1.6.0' && GO111MODULE=off go install github.com/onsi/ginkgo/ginkgo
```
[the remark this answers is not in the document]
Comment (sebbonnet) [reply cassandra-snapshot/Makefile]: same comment as above about upgrading to `v1.7.0`

File: cassandra-operator/hack/pin-dependency.sh (hunk 1 above)
Comment (sebbonnet) [reply cassandra-operator/hack/pin-dependency.sh]: Good point. Is this also needed for cassandra-snapshot?

File: cassandra-operator/Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (sebbonnet) [reply cassandra-operator/Makefile]: actually we need `v1.7.0` which added compatibilty for go modules

**Comment by kragniz** [comment]
Build succeeded: https://travis-ci.com/sky-uk/cassandra-operator/builds/111772697

Removing the temporary pin on https://github.com/sky-uk/licence-compliance-checker/pull/19

File: cassandra-snapshot/go.mod [hunk 5]
```diff
@@ -0,0 +1,41 @@
+module github.com/sky-uk/cassandra-operator/cassandra-snapshot
+
+go 1.11
```
[the remark this answers is not in the document]
Comment (kragniz) [reply cassandra-snapshot/go.mod]: This is supposed to denote the minimum go version required for this package. I can update this to 1.12, or just remove it if you'd like.

See https://github.com/golang/go/issues/28221

[the remark this answers is not in the document]
Comment (kragniz) [reply cassandra-snapshot/go.mod]: I updated it

**Comment by kragniz** [comment]
@sebbonnet I've added a validation step before the go-build target

**Pull request merged**